synapse/config/cas.py

```diff
@@ -41,6 +41,13 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
             public_baseurl = self.root.server.public_baseurl
             self.cas_service_url = public_baseurl + "_matrix/client/r0/login/cas/ticket"
 
+            self.cas_protocol_version = cas_config.get("protocol_version")
+            if self.cas_protocol_version not in [1, 2, 3]:
```

Comment:
This will by default break current configurations (which will default to `None`). I think we want:

```suggestion
            if self.cas_protocol_version is not None and self.cas_protocol_version not in [1, 2, 3]:
```